[Cheam Jia Wei] iP

Cheamybunny opened the pull request:
### DukePro
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,

- Text-based
- Easy to learn
- ~FAST~ SUPER FAST to use

All you need to do is,

1. Download it from [here](https://github.com/Cheamybunny/ip)
2. Double-click it
3. Add your tasks
4. Let it manage your tasks for you :wink:

And it is **FREE!**

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

***********
If you are a Java programmer, you can use it to practice Java too. Here is the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

```

Commit:
Add Gradle support

Cheamybunny committed:
Level-1

Cheamybunny committed:
Level-2

Cheamybunny committed:
Level-3

Cheamybunny committed:
Level-4

Cheamybunny committed:
A-TextUiTesting

Cheamybunny committed:
Level-5

Cheamybunny committed:
Level-6

Cheamybunny committed:
A-Enums

Cheamybunny committed:
Level-7

Cheamybunny committed:
Level-8

Cheamybunny committed:
Level-8 merged

Cheamybunny committed:
A-MoreOOP & A-Packages

Cheamybunny committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Cheamybunny committed:
A-Gradle

Cheamybunny committed:
A-JUnit

Cheamybunny committed:
A-JavaDoc

Cheamybunny committed:
test

Cheamybunny committed:
A-JavaDoc

Cheamybunny committed:
A-CodingStandard

Cheamybunny committed:
Level-9

Cheamybunny committed:
Merge branch 'branch-A-JavaDoc'

Cheamybunny committed:
Resolved merge conflict

Cheamybunny committed:
Resolved merge conflict with branch-Level-9

Cheamybunny committed:
Level-10

Cheamybunny committed:
Merge branch 'branch-Level-10'

jiexuanc commented on src/main/java/duke/commands/Command.java:
```diff
@@ -0,0 +1,29 @@
+     * @param taskList The TaskList that will be modified or accessed.
+     * @param inter The Ui that will interact with the user.
+     * @param store The storage that will help store the task into the data file if TaskList is modified.
+     */
```
Might be better to include @return for execute method, not obvious at first glance that the return value is the message to be printed

jiexuanc commented on src/main/java/duke/commands/Command.java:
```diff
@@ -0,0 +1,29 @@
+     * @param inter The Ui that will interact with the user.
+     * @param store The storage that will help store the task into the data file if TaskList is modified.
+     */
+    public abstract String execute(TaskList taskList, Ui inter, Storage store);
```
I think it might be better to include @return for execute method, not obvious at first glance what the return value is meant to represent.

jiexuanc commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,100 @@
+    /**
+     * Message to greet the user.
+     */
+    public String greet() {
```
I think having @return would better adhere to coding standards although it seems rather redundant.

jiexuanc commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,97 @@
+        String[] details = input.split(" ", 2);
+        Word w = Word.valueOf(details[0].toUpperCase());
+
+        switch (w) {
```
Good job that the correct indentation for switch case is practised!

jiexuanc commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,97 @@
+public class Parser {
+    private enum Word {
+        BYE, LIST, MARK, UNMARK, TODO, DEADLINE, EVENT, DELETE, THROUGH, FIND
+    }
```
Great use of enum.

jiexuanc reviewed (commented):
Overall, wonderful code that adheres to the coding standard.

Zeli0 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,223 @@
+     * @param date The data and time the user is checking.
+     * @return A string listing the tasks occurring through the date and time.
+     */
+    public String through(LocalDateTime date) {
```
I personally think that the method name "through" is a bit ambiguous and I didn't immediately get it before reading the javadoc. Would it be possible to give it a slightly more intuitive name?

Zeli0 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,52 @@
+     */
+    public String getResponse(String input) {
+        try {
+            Command c = Parser.parse(input);
```
I like how you SLAP here! Makes everything very clean.

Zeli0 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,223 @@
+     * @param input String that has been input by the user contain the details of the task.
+     * @return The newly created Todos task that was added into the TaskList.
+     */
+    public Task todo(String input) {
```
I think your task creation methods could have a bit more information as to what they specifically do. Maybe you can add something like "makeNewTodo" instead?

Zeli0 commented on src/main/java/duke/commands/FindCommand.java:
```diff
@@ -0,0 +1,26 @@
+        this.input = input;
+    }
+
+    public String execute(TaskList taskList, Ui inter, Storage store) {
```
I like your use of KISSing here! Very simple to understand.

Zeli0 reviewed (commented):
Overall your code is great! I have much to learn from you. Just have a few suggestions.

Cheamybunny committed:
A-Assertions

Cheamybunny committed:
A-CodeQuality

Cheamybunny committed:
Merge pull request #2 from Cheamybunny/branch-A-Assertions

A-Assertions

Cheamybunny committed:
Merge branch 'master' into branch-A-CodeQuality

Cheamybunny committed:
Merge pull request #3 from Cheamybunny/branch-A-CodeQuality

A-CodeQuality

Cheamybunny committed:
Merge branch 'master' of https://github.com/Cheamybunny/ip

Cheamybunny committed:
BCD-Extension C-Update Added

Tasks are not able to be updated. Had to be deleted and re-added into the list.

Let's add the feature to update tasks with the neccessary details provided to reduce
the need of doing extra steps in the TaskList.

Cheamybunny committed:
Merge branch 'branch-BCD-Extension'

Cheamybunny committed:
A-BetterGui

Cheamybunny committed:
Merge branch 'master' of https://github.com/Cheamybunny/ip

Cheamybunny committed:
A-UserGuide

Cheamybunny committed:
Update README.md
